deploy/helm/sumologic/conf/setup/setup.sh

```diff
@@ -83,6 +135,52 @@ function should_create_fields() {
     fi
 }
 
+function pvc_test_create_statefulset() {
+    kubectl create namespace ${TEST_PVC_NAMESPACE}
+    echo "${TEST_PVC_STATEFULSET_CONFIG}" > ${TEST_PVC_FILE}
+
+    kubectl -n ${TEST_PVC_NAMESPACE} apply -f ${TEST_PVC_FILE}
+    if [[ $? -eq 0 ]]; then
+      return 0
+    else
+      echo "Creating statefulset failed"
+      pvc_test_cleanup
+      exit 1
+    fi
+}
+
+function pvc_test_check() {
+    local pending_pvcs
+    pending_pvcs=$(kubectl -n ${TEST_PVC_NAMESPACE} get pvc | grep -c  "Pending")
+    readonly pending_pvcs
+
+    if [[ $? -ne 0 ]]; then
+        echo "Querying pvcs failed"
+        exit 1
+    else
+        return $pending_pvcs
+    fi
+}
+
+function pvc_test_cleanup() {
+    kubectl delete all --all -n ${TEST_PVC_NAMESPACE}
+    kubectl delete pvc --all -n ${TEST_PVC_NAMESPACE}
+    rm ${TEST_PVC_FILE}
+}
+
+function can_create_pvc() {
+    pvc_test_create_statefulset
+    sleep 30
+    pvc_test_check
```

Comment:
Not sure what constant should be used here.

Reply:
do we need to wait 30 seconds? How do we know pvc cannot be created automatically?

Reply:
Afaik the only way to do it is to try to create a pvc and see if it gets created - and it doesn't happen automatically. Although it's not perfect, I think it's fine as long as we don't get false negatives (which we might, if creating a pvc takes longer than 30 seconds).
We can lower the impact of waiting by moving this to a separate process, run at the beginning of the script.